brave/browser/brave_browser_context.cc

```diff
@@ -643,6 +751,23 @@ void BraveBrowserContext::SetExitType(ExitType exit_type) {
   }
 }
 
+void BraveBrowserContext::SetTorNewIdentity(const GURL& url,
```

Comment:
what is this used for?

Reply:
it's tied to a button that the user clicks when they want a new Tor circuit. https://tb-manual.torproject.org/en-US/managing-identities.html

Reply:
this doesn't seem to match the described functionality - "This option is useful if you want to prevent your subsequent browser activity from being linkable to what you were doing before. Selecting it will close all your open tabs and windows, clear all private information such as cookies and browsing history, and use new Tor circuits for all connections"

Reply:
it looks like it just creates a new tor proxy, but based on the description I think each Tor identify would be similar to a session tab and create a new browser context

Reply:
Sorry, it should be this `New Tor Circuit for this Site`

Reply:
if we're trying to make this friendly for non-technical users I don't think `New Tor Circuit for this site` is going to help. I had to dig a little bit to understand exactly what this does. Why don't we just do this anytime a user reloads a url (explicit reload vs normal navigation)?

Reply:
We've already used different tor circuits for different sites. This function enable users to change it.
cc @riastradh-brave

Reply:
The user-visible operation was just copied from the Tor Browser.  It has not undergone any UX review on the Brave side, which certainly ought to happen before this is released.  However, I expect that whatever manifestation it has to the user, we will probably want an operation to require a new Tor circuit for an origin -- possibly never used on its own, always in conjunction with other things, but I can't see that far ahead right now.